api/runtime/hooks/v1alpha1/inplaceupdate_types.go

```diff
@@ -77,17 +66,28 @@ type CanUpdateMachineResponse struct {
 	// CommonResponse contains Status and Message fields common to all response types.
 	CommonResponse `json:",inline"`
 
-	// machinePatch, when applied to the current Machine spec, indicates changes handled in-place.
+	// machinePatch when applied to the current Machine spec, indicates changes handled in-place.
 	// +optional
-	MachinePatch *Patch `json:"machinePatch,omitempty"`
+	MachinePatch Patch `json:"machinePatch,omitempty,omitzero"`
```

Comment:
We don't need the pointers anymore, omitzero handles correct marshalling (also matches regular API conventions for CRD API types)

Also reduces potential for nil pointers

Reply:
I think the short answer is, the trade-offs are a bit different for CRDs. I'm not sure about the latest state of the doc but it's mostly focused on builtin types

We were discussion this with Joel when we refactored everything with v1beta2 and this is inline with what we did there

(just to give some additional context)